107/src/main/java/org/ehcache/jsr107/internal/Jsr107CacheConfigurationParser.java

```diff
@@ -69,4 +71,14 @@ public URI getNamespace() {
           fragment.getTagName(), (fragment.getParentNode() == null ? "null" : fragment.getParentNode().getLocalName())));
     }
   }
+
+  @Override
+  public Class<Jsr107Service> getServiceType() {
+    return null;
+  }
+
+  @Override
+  public Node marshalServiceConfiguration(Document doc, ServiceConfiguration<Jsr107Service> serviceConfiguration) {
```

Comment:
Why are you passing the whole document to the method?  It doesn't need (and indeed shouldn't have) access to anything else about the configuration.

Reply:
Also, isn't it required that this return an element, a Node isn't precise enough.

Reply:
We were getting problem if the Document object was not passed from the caller. If a document object was not passed and it was being created inside the method (to create an element from the document), then it was getting error; reference to elements in ehcache-core.xsd was not getting visible. We found that while writing the unit tests. 

Node is more general. That's why I put it in the return signature. Can change it to Element

Reply:
The method you are looking for is: Document.importNode(...)